### PR TITLE
feat: static analysis for browser rendered preview projects

### PR DESCRIPTION
**Pull request by karishnu**
## Summary
Adds in-memory static analysis capability for browser-rendered preview projects that don't have a sandbox environment. This enables code quality checks (syntax validation, HTML/CSS cross-validation) without requiring sandbox infrastructure.

## Changes
- **`worker/agents/core/behaviors/base.ts`**: Added conditional logic to use in-memory analysis for browser-rendered projects (based on `renderMode === 'browser'`), and enabled static analysis in `fetchAllIssues()` for projects without sandbox
- **`worker/agents/prompts.ts`**: Enhanced `serializeStaticAnalysis()` to format issues when no raw output is available, with truncation support
- **`worker/services/static-analysis/`**: New static analysis service with:
  - `InMemoryAnalyzer`: Main orchestrator that coordinates language analyzers and cross-file validators
  - `JavaScriptAnalyzer`: Syntax validation using acorn parser
  - `HTMLAnalyzer`: Tag matching validation using htmlparser2
  - `CSSAnalyzer`: Brace/string/comment balancing validation
  - `HTMLCSSCrossValidator`: Cross-file validation for undefined CSS classes/IDs referenced in HTML

## Motivation
Browser-rendered projects (static HTML/CSS/JS) don't spin up a sandbox environment, which previously meant no static analysis was available. This PR enables lightweight, in-memory code quality checks for these projects using parsing libraries already available in the worker runtime.

## Testing
- Test with a browser-rendered template project to verify static analysis runs
- Verify sandbox-based projects still use the existing analysis path
- Test edge cases: malformed HTML, unclosed CSS braces, JS syntax errors, undefined CSS classes

## Breaking Changes
None - this is additive functionality that only affects browser-rendered projects.

### CODE REVIEW

File: worker/agents/core/behaviors/base.ts [hunk 1]
```diff
@@ -685,8 +718,11 @@ export abstract class BaseCodingBehavior<TState extends BaseProjectState>
     }
 
     async fetchAllIssues(resetIssues: boolean = false): Promise<AllIssues> {
+        // For browser-rendered projects (no sandbox), only run static analysis
         if (!this.state.sandboxInstanceId) {
-            return { runtimeErrors: [], staticAnalysis: { success: false, lint: { issues: [], }, typecheck: { issues: [], } } };
+            const staticAnalysis = await this.runStaticAnalysisCode();
```
Comment (AshishKumar4) [inline worker/agents/core/behaviors/base.ts:726]: When sandboxInstanceId isn't available, we cannot directly assume that it's a browser rendered app
